### ISSUE TITLE
Externalize message catch / conditional as job

### ISSUE THREAD

**Issue by jonathanlukas** [camunda/camunda#15898]
**Is your feature request related to a problem? Please describe.**

See https://github.com/camunda/zeebe/issues/14732#issuecomment-1764117225

TL;DR: Inbound connectors need to correlate the results of their subscriptions and publish a message.

Instead, the handling of a message catch event / conditional event could be externalized completely.

This would allow for more flexibility in the implementation:

* publishing a message might make sense and you can rely on this mechanism very often, especially when the subscription has a lifecycle independent of process instances or even element instances. Here, the buffer is a killer feature
* sometimes, I would like to have an implementation just like the inbound connector has (especially if I create an inbound connector), where my subscription is scoped to an elementInstance

**Describe the solution you'd like**

Every message catch event, receive task and conditional event should have a selection for implementation where job can be selected.

For a message, the implementation could be part of the `<bpmn:message/>` and would be shared among all events referening it.

Then, a subscription would be opened by the job worker implementing the job type. Thanks to https://docs.camunda.io/docs/apis-tools/zeebe-api/gateway-service/#updatejobtimeout-rpc, a subscription can be kept open as long as required.

**Describe alternatives you've considered**
no

**Additional context**
no

**Comment by npepinpe** [comment]
ZDP triage: sounds like a FR for the engine, so we'll remove it from our board 👍

**Comment by tmetzke** [comment]
ZPA triage:
- This generally makes sense in the scope of the product.
- @aleksander-dytko, please decide on the priority of this, also concerning https://github.com/camunda/zeebe/issues/14732
